### PR TITLE
fix: refactor connection failed description

### PR DESCRIPTION
**Pull request by DorraJaouad**
### ☑️ Resolves

* Fix part of #13160

### 🖼️ Screenshots / Screencasts

🏚️ Before | 🏡 After
-- | --
Screenshot 2024-08-30 111330 | Screenshot 2024-08-30 111116
| Before | Screenshot 2024-08-30 112219 |

### 🏁 Checklist

- [ ] 🌏 Tested with different browsers / clients:
  - [x] Chromium (Chrome / Edge / Opera / Brave)
  - [ ] Firefox
  - [ ] Safari
  - [ ] Talk Desktop
  - [ ] Not risky to browser differences / client
- [ ] 🖌️ Design was reviewed, approved or inspired by the design team
- [ ] ⛑️ Tests are included or not possible
- [ ] 📗 User documentation in https://github.com/nextcloud/documentation/tree/master/user_manual/talk has been updated or is not required

### CODE REVIEW

File: src/components/CallView/shared/EmptyCallView.vue [hunk 1]
```diff
@@ -148,19 +160,36 @@ export default {
...
+					errorReason = t('spreed', "Lobby is still active and you're not a moderator")
+					break
+				default:
+					if (this.connectionFailed.data?.error) {
+						errorReason = this.connectionFailed.data.error
+					} else {
+						errorReason = t('spreed', 'Unknown error occured')
+					}
+				}
+				return errorReason
```
Comment (ShGKme) [inline src/components/CallView/shared/EmptyCallView.vue:192]: To get rid of switch:

```suggestion
				const statusErrors = {
					0: t('spreed', 'No response from server'),
					400: t('spreed', 'Recording consent is required'),
					403: t('spreed', 'This conversation is read-only'),
					404: t('spreed', 'Conversation not found or not joined'),
					412: t('spreed', "Lobby is still active and you're not a moderator"),
				}
				const statusCode = this.connectionFailed.meta?.statuscode
				return statusCode 
					? statusErrors[statusCode]
					: this.connectionFailed.data?.error
						|| t('spreed', 'Unknown error occured')
```

or

```suggestion
				const statusErrors = {
					0: t('spreed', 'No response from server'),
					400: t('spreed', 'Recording consent is required'),
					403: t('spreed', 'This conversation is read-only'),
					404: t('spreed', 'Conversation not found or not joined'),
					412: t('spreed', "Lobby is still active and you're not a moderator"),
				}

				const statusCode = this.connectionFailed.meta?.statuscode
				if (statusErrors[statusCode]) {
					return statusErrors[statusCode]
				}

				if (this.connectionFailed.data?.error) {
					return this.connectionFailed.data.error
				}
				
				return t('spreed', 'Unknown error occured')
```

`switch` works as a many `goto` wwith labels, being huge, with a requirement of `break` (that can be lost) and less declarative.

Comment (ShGKme) [reply src/components/CallView/shared/EmptyCallView.vue:192]: BTW, can status code be 0 here?

File: src/components/CallView/shared/EmptyCallView.vue [hunk 2]
```diff
@@ -8,8 +8,9 @@
 		:class="{
 			'empty-call-view--sidebar': isSidebar,
 			'empty-call-view--small': isSmall
-		}">
-		<div class="icon" :class="iconClass" />
+		}"
+		data-theme-dark>
+		<component :is="emptyCallViewIcon" :size="isSidebard? 32 : 64" class="empty-call-view__icon" />
```
Comment (ShGKme) [inline src/components/CallView/shared/EmptyCallView.vue:13]:
```suggestion
		<component :is="emptyCallViewIcon" :size="isSidebar ? 32 : 64" class="empty-call-view__icon" />
```

File: src/components/CallView/shared/EmptyCallView.vue (hunk 1 above)
Comment (DorraJaouad) [reply src/components/CallView/shared/EmptyCallView.vue:192]: iOS is taking it into account, haven't checked if there is indeed such a response
https://github.com/nextcloud/talk-ios/blob/81ecc92ada12ae88f9a2f2aa6c297fa5295111d8/NextcloudTalk/NCCallController.m#L230-L257

File: src/components/CallView/shared/EmptyCallView.vue [hunk 3]
```diff
@@ -8,8 +8,9 @@
 		:class="{
 			'empty-call-view--sidebar': isSidebar,
 			'empty-call-view--small': isSmall
-		}">
-		<div class="icon" :class="iconClass" />
+		}"
+		data-theme-dark>
+		<component :is="emptyCallViewIcon" :size="isSidebar? 32 : 64" class="empty-call-view__icon" />
```
Comment (Antreesy) [inline src/components/CallView/shared/EmptyCallView.vue:13]:
```suggestion
		<component :is="emptyCallViewIcon" :size="isSidebar ? 32 : 64" class="empty-call-view__icon" />
```

File: src/components/CallView/shared/EmptyCallView.vue [hunk 4]
```diff
@@ -28,9 +29,15 @@
 </template>
 
 <script>
+import AccountMultipleIcon from 'vue-material-design-icons/AccountMultiple.vue'
```
Comment (Antreesy) [inline src/components/CallView/shared/EmptyCallView.vue:32]: Grigorii started to follow this naming style, maybe let's keep it?
```suggestion
import IconAccountMultiple from 'vue-material-design-icons/AccountMultiple.vue'
```

File: src/components/CallView/shared/EmptyCallView.vue [hunk 5]
```diff
@@ -148,19 +160,27 @@ export default {
...
 		},
 
 		message() {
-			if (this.connectionFailed === 'consent') {
-				return t('spreed', 'Recording consent is required')
-			}
-
 			if (this.connectionFailed) {
-				return t('spreed', 'Something went wrong')
+				const statusErrors = {
```
Comment (Antreesy) [inline src/components/CallView/shared/EmptyCallView.vue:168]: Shall we move it out of `export default {` component block?

File: src/components/CallView/shared/EmptyCallView.vue [hunk 6]
```diff
@@ -148,19 +160,27 @@ export default {
...
+					0: t('spreed', 'No response from server'),
+					400: t('spreed', 'Recording consent is required'),
+					403: t('spreed', 'This conversation is read-only'),
+					404: t('spreed', 'Conversation not found or not joined'),
+					412: t('spreed', "Lobby is still active and you're not a moderator"),
+				}
+				const statusCode = this.connectionFailed.meta?.statuscode
```
Comment (Antreesy) [inline src/components/CallView/shared/EmptyCallView.vue:175]: Either this or ensure that getter always returns an object
```suggestion
				const statusCode = this.connectionFailed?.meta?.statuscode
```

File: src/components/CallView/shared/EmptyCallView.vue (hunk 4 above)
Comment (DorraJaouad) [reply src/components/CallView/shared/EmptyCallView.vue:32]: We were using "...Icon" ending too, which one to keep ?

Comment (Antreesy) [reply src/components/CallView/shared/EmptyCallView.vue:32]: I'd migrate all later to have prefix `Icon...`, same as `Nc...` components. Then we don't need to group them later under `// Icons` comment

File: src/components/CallView/shared/EmptyCallView.vue (hunk 1 above)
Comment (SystemKeeper) [reply src/components/CallView/shared/EmptyCallView.vue]: > iOS is taking it into account, haven't checked if there is indeed such a response

Status code 0 is an internal code from iOS, it is not a valid HTTP status code. On iOS it occurs on various reasons, for example timeout.

Comment (Antreesy) [reply src/components/CallView/shared/EmptyCallView.vue]: Then I'd get rid of it, as we have `Unkown error occured` (Wasn't it `Something went wrong?`)

File: src/components/CallView/shared/EmptyCallView.vue (hunk 4 above)
Comment (ShGKme) [reply src/components/CallView/shared/EmptyCallView.vue:32]: "Icon*" follows a new convention on the server, groupware apps and libs. It also follows Bye style guide - related components should have a common prefix, not a suffix, so they are together alphabetically

File: src/components/CallView/shared/EmptyCallView.vue [hunk 7]
```diff
@@ -148,18 +166,19 @@ export default {
...
-			if (this.connectionFailed === 'consent') {
-				return t('spreed', 'Recording consent is required')
-			}
-
 			if (this.connectionFailed) {
+				const statusCode = this.connectionFailed?.meta?.statuscode
+				if (STATUS_ERRORS[statusCode]) {
+					return STATUS_ERRORS[statusCode]
+				}
+				if (this.connectionFailed.data?.error) {
```
Comment (Antreesy) [inline src/components/CallView/shared/EmptyCallView.vue:178]:
```suggestion
				if (this.connectionFailed?.data?.error) {
```

**Review by Antreesy — APPROVED** [review]
Tested

**Pull request merged**